### DISCUSSION TITLE
I'm trying to solve Error: jQuery requires a window with a document

### DISCUSSION THREAD

**Discussion by guddukmr786** [vercel/next.js#16433] [Help] [answered]
Error I'm getting - 
error - ./node_modules/jsdom/lib/jsdom/living/xhr/XMLHttpRequest-impl.js
Module not found: Can't resolve 'child_process' in '/var/www/html/nmpwanext/node_modules/jsdom/lib/jsdom/living/xhr'

I'm trying to import var $ = require('jquery')(require("jsdom").jsdom().parentWindow); or var $ = require('jquery')(require("jsdom").JSDOM().window);
This is working perfect without **nextJS** ....
but with **nextJS** not working

my code where I'm using this functionality 
createtweetPoints = () => {
        var count = 1;
        $('.twitterPoint').each(
            function () {
                if (count > 3) {
                    count = 1;
                }
                var tweet_text = "";

                tweet_text = $(this).text();
                var html = "";
                html = html + '<div class=twitterContaoner>';
                html = html + '<div class=twitt_content>';
                html = html + '<blockquote class="customQuote" id="blkq-' + count + '"> ' + tweet_text + ' </blockquote>';
                html = html + '</div>';
                html = html + '</div>';
                html = html + '</div>';
                $(this).removeClass('twitterPoint');
                $(this).text('');
                $(this).removeAttr('style');
                $(this).addClass('twitter_div twitter_g' + count);
                $(this).append(html);
                count++;

            });

    }

Guy's please help me to out from this situation

**Comment by jamesmosier** [comment, marked as answer]
You could use `next/dynamic` with `ssr: false` to import the component that uses jQuery (or `window`). https://nextjs.org/docs/advanced-features/dynamic-import#with-no-ssr

`next/dynamic` only works with React components, so you can put your logic within a component and then use `ssr: false` when you import that component.

**Reply by guddukmr786** [reply]
Thank you so much @jamesmosier
